librz/bin/bin.c

```diff
@@ -673,6 +673,33 @@ RZ_API RZ_BORROW RzBinMap *rz_bin_object_get_map_at(RZ_NONNULL RzBinObject *o, u
 	return NULL;
 }
 
+/**
+ * \brief Find the binary symbol at offset \p off.
+ *
+ * This function returns the binary symbol that contains offset \p off.
+ * It iterates over the list of symbols and directly compares the offset with the virtual or physical address.
+ *
+ * \param o Reference to the \p RzBinObject instance
+ * \param off Address to search
+ * \param va When false, the offset \p off is considered a physical address; otherwise, a virtual address
+ * \return Pointer to a \p RzBinSymbol containing the address, or NULL if no symbol is found at the address
+ */
+RZ_API RZ_BORROW RzBinSymbol *rz_bin_object_get_symbol_at(RZ_NONNULL RzBinObject *o, ut64 off, bool va) {
+	rz_return_val_if_fail(o, NULL);
+
+	RzBinSymbol *sym;
+	RzListIter *iter;
+
+	rz_list_foreach (o->symbols, iter, sym) {
+		if (va && off == sym->vaddr) {
```

Comment:
Checking the `va` value inside the loop adds unnecessary instructions every time. Just check it outside the loop to speed up things in case there are a lot of symbols.

Reply:
I've done and made the changes requested, you can take a look.